### PR TITLE
Update module

### PR DESCRIPTION
**Pull request by jcriadop**
- Multi-label order generator (filtered by date)
- Better update/upgrade module process
- Voucher discount compatibility fixed
- Tax policies fixed
- Minor updates and bug fixed
- Added link to configuration when module is not installed

### CODE REVIEW

File: AdminSeur.php [hunk 1]
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
...
 	  </fieldset>
 
-	  </div>';
+	  </div>
+				';
+
+	}
+	public function getAllOrders($id = null, $reference = null, $date_start = null, $date_end = null, $name = null, $address = null, $postal_code = null, $city  = null, $state_name = null, $country = null)
+	{
+		$sql = 'SELECT o.reference,o.id_order,o.date_add,o.reference, a.firstname, a.lastname, a.address1, a.address2, a.postcode, a.city, cl.name as country, s.name as state FROM '._DB_PREFIX_.'orders o INNER JOIN '._DB_PREFIX_.'seur_order so ON so.id_order  = o.id_order  INNER JOIN '._DB_PREFIX_.'address a ON o.id_address_delivery = a.id_address INNER JOIN '._DB_PREFIX_.'country_lang cl ON cl.id_country = a.id_country AND cl.id_lang ='.$this->context->language->id.' LEFT JOIN '._DB_PREFIX_.'state s ON s.id_state = a.id_state ';
```
Comment (Quetzacoalt91) [inline AdminSeur.php]: Please escape `$this->context->language->id` with an `(int)` cast

File: AdminSeur.php [hunk 2]
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
...
+		$where = array();
+		if ((int)$id > 0)
+			$where[] = ' o.id_order ='.$id.' ';
```
Comment (Quetzacoalt91) [inline AdminSeur.php]: Please escape the data stored in $id

File: AdminSeur.php [hunk 3]
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
...
+		
+		if (trim($reference) != '')
+			$where[] = ' o.reference like \''.$reference.'\' ';
```
Comment (Quetzacoalt91) [inline AdminSeur.php]: Please escape the data stored in $reference

File: AdminSeur.php [hunk 4]
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
...
+		
+		
+		if (trim($date_start) != '' && trim($date_end) != '' )
+			$where[] = ' o.date_add between \''.$date_start.'\' AND   \''.$date_end.'\' ';
```
Comment (Quetzacoalt91) [inline AdminSeur.php]: Please escape the data stored in your dates

File: AdminSeur.php [hunk 5]
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
...
+			
+		if (trim($date_start) != '' && trim($date_end) == '' )
+			$where[] = ' o.date_add >= \''.$date_start.'\'  ';
```
Comment (Quetzacoalt91) [inline AdminSeur.php]: Please escape the data stored in $date_start

File: AdminSeur.php [hunk 6]
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
...
+			
+		if (trim($date_start) == '' && trim($date_end) != '' )
+			$where[] = ' o.date_add <= \''.$date_start.'\'  ';
```
Comment (Quetzacoalt91) [inline AdminSeur.php]: Please escape the data stored in $date_start

File: AdminSeur.php [hunk 7]
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
...
+			
+		if (trim($name) != ''  )
+			$where[] = ' ( a.firstname like = \'%'.$name.'%\'  OR a.lastname like = \'%'.$name.'%\'  )  ';
```
Comment (Quetzacoalt91) [inline AdminSeur.php]: Please escape the data stored in $name

File: AdminSeur.php [hunk 8]
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
...
+		
+		if (trim($address) != ''  )
+			$where[] = ' ( a.address1 like = \'%'.$address.'%\'  OR a.address2 like = \'%'.$address.'%\'  )  ';
```
Comment (Quetzacoalt91) [inline AdminSeur.php]: Please escape the data stored in $address

File: AdminSeur.php [hunk 9]
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
...
+		
+		
+		if (trim($postal_code) != ''  )
+			$where[] = ' a.post_code like = \'%'.$postal_code.'%\'   ';
```
Comment (Quetzacoalt91) [inline AdminSeur.php]: Please escape the data stored in $postal_code

File: AdminSeur.php [hunk 10]
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
...
+		
+		if (trim($city) != ''  )
+			$where[] = ' a.city like = \'%'.$city.'%\'   ';
```
Comment (Quetzacoalt91) [inline AdminSeur.php]: Please escape the data stored in $city

File: AdminSeur.php [hunk 11]
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
...
+		
+		if (trim($state_name) != ''  )
+			$where[] = ' a.state like = \'%'.$state_name.'%\'   ';
```
Comment (Quetzacoalt91) [inline AdminSeur.php]: Please escape the data stored in $state_name

File: AdminSeur.php [hunk 12]
```diff
@@ -417,14 +425,368 @@ public function displayFormDeliveries($response = null, $detail = null, $error =
...
+		
+		if (trim($country) != ''  )
+			$where[] = ' a.country like = \'%'.$country.'%\'   ';
```
Comment (Quetzacoalt91) [inline AdminSeur.php]: Please escape the data stored in $country

**Comment by Quetzacoalt91** [comment]
Hi,

jQuery, jQuery UI, jQuery UI datepicker already exist on PrestaShop. Please use them instead of including them a second time in your module.
BTW, the validation will be really simpler if you do this, because the diff will be lighter and with the same functionalities (Currently, I cannot check all your changes with the truncated diff).

Thanks

**Comment by Quetzacoalt91** [comment]
Hi,

Your controller are are made to load all the data needed by a template. You should not use it to create the HTML code.
For instance, please move all the HTML content found in _controllers/admin/AdminSeurController.php_ in a tpl file. If another files are also concerned, please do the same work.

Thanks

File: AdminSeur.php (hunk 12 above)
Comment (jcriadop) [inline AdminSeur.php]: Solved.

File: AdminSeur.php (hunk 11 above)
Comment (jcriadop) [inline AdminSeur.php]: Solved.

File: AdminSeur.php (hunk 10 above)
Comment (jcriadop) [inline AdminSeur.php]: Solved.

File: AdminSeur.php (hunk 9 above)
Comment (jcriadop) [inline AdminSeur.php]: Solved

File: AdminSeur.php (hunk 8 above)
Comment (jcriadop) [inline AdminSeur.php]: Solved.

File: AdminSeur.php (hunk 7 above)
Comment (jcriadop) [inline AdminSeur.php]: Solved.

File: AdminSeur.php (hunk 6 above)
Comment (jcriadop) [inline AdminSeur.php]: Solved.

File: AdminSeur.php (hunk 5 above)
Comment (jcriadop) [inline AdminSeur.php]: Solved.

File: AdminSeur.php (hunk 4 above)
Comment (jcriadop) [inline AdminSeur.php]: Solved.

File: AdminSeur.php (hunk 3 above)
Comment (jcriadop) [inline AdminSeur.php]: Solved.

File: AdminSeur.php (hunk 2 above)
Comment (jcriadop) [inline AdminSeur.php]: Solved.

File: AdminSeur.php (hunk 1 above)
Comment (jcriadop) [inline AdminSeur.php]: Solved.

**Comment by jcriadop** [comment]
Hi, new commit solve this.

**Comment by jcriadop** [comment]
Hi, new commit solve this.

**Comment by Quetzacoalt91** [comment]
Thanks a lot, the PRs is really easier to review now. :smiley:

File: install/seurcashondelivery/seurcashondelivery.php [hunk 13]
```diff
@@ -503,34 +546,40 @@ public function validateOrderFORWEBS_v5($id_cart, $id_order_state, $amount_paid,
 					$order->gift = (int)$this->context->cart->gift;
 					$order->gift_message = $this->context->cart->gift_message;
 					$order->conversion_rate = $this->context->currency->conversion_rate;
+					
 					$amount_paid = !$dont_touch_amount ? Tools::ps_round((float)$amount_paid, 2) : $amount_paid;
 					$order->total_paid_real = 0;
 
 					$order->total_products = (float)$this->context->cart->getOrderTotal(false, Cart::ONLY_PRODUCTS, $order->product_list, $id_carrier);
 					$order->total_products_wt = (float)$this->context->cart->getOrderTotal(true, Cart::ONLY_PRODUCTS, $order->product_list, $id_carrier);
+					$order->total_products_wt = (float)$this->context->cart->getOrderTotal(true, Cart::ONLY_PRODUCTS, $order->product_list, $id_carrier);
```
Comment (Quetzacoalt91) [inline install/seurcashondelivery/seurcashondelivery.php]: Be careful, you duplicated the previous line

**Comment by Quetzacoalt91** [comment]
### ajax/getLabelAjax.php
- SQL queries are not escaped

### config.xml
- should be in english

### Localized config_en.xml & config_fr.php
- Should be deleted

### Unused translation files to remove
- en.php
- fr.php

### Are these template files used ?
- views/templates/admin/header.tpl
- views/templates/admin/backoffice_header.tpl

### Validator comments to check
- Warning & errors increased from 654/559 to 1089/1099. Please have a look on the report

**Comment by jcriadop** [comment]
Hi, All solved on last commit, except "Localized config_en.xml & config_fr.php", I don't find these files.

**Comment by Quetzacoalt91** [comment]
Sorry, I was talking about config_es.xml !

**Comment by jcriadop** [comment]
Do you want I remove config_es.xml ??

**Comment by Quetzacoalt91** [comment]
Yes please , we only keep config.xml

**Comment by jcriadop** [comment]
Done, we removed.

**Comment by Quetzacoalt91** [comment]
Hi,

I'm still waiting for a new commit, did you forget to push ?

**Comment by jcriadop** [comment]
Hi, I sent last commit 2 days ago with last reques:

https://github.com/jcriadop/seur/commit/de2eb13e6f8e9b9b4e3424a8829155fdd8d78cb4

I don't see any question pending for me.

**Comment by Quetzacoalt91** [comment]
But config_es.xml in https://github.com/PrestaShop/seur/pull/1/files is still here. That's why

**Comment by jcriadop** [comment]
Ok, commit sent

**Comment by Quetzacoalt91** [comment]
Thanks, the PR has just been merged and the changes should be available from tomorrow.

Best regards

File: controllers/admin/AdminSeur15Controller.php [hunk 14]
```diff
@@ -29,50 +29,481 @@
...
+require_once(_PS_MODULE_DIR_.'seur/classes/Expedition.php');
 
 class AdminSeur15Controller extends ModuleAdminController {
 
+	
+	public function __construct()
+	{
+	parent::__construct();
+	$this->module = Module::getInstanceByName('seur');
+	}
```
Comment (Quetzacoalt91) [inline controllers/admin/AdminSeur15Controller.php:41]: This part is already executed by the parent class

File: install/seurcashondelivery/seurcashondelivery.php [hunk 15]
```diff
@@ -307,7 +306,36 @@ public function hookPaymentReturn($params)
 				return $this->display(__FILE__, 'views/templates/hook/confirmation.tpl');
     }
 
-	public function getCargo($cart, $check_id_currency = true)
+	public function getCargo($order, $check_id_currency = true)
+	{
+
+		if ((int)$order->id >0)
+		{
+			$sql = 'SELECT codfee FROM '._DB_PREFIX_.'seur_order WHERE id_order='.$order->id;
```
Comment (Quetzacoalt91) [inline install/seurcashondelivery/seurcashondelivery.php:314]: You cast the variable in your test, but not in the SQL request. Please add a `(int)` cast

File: install/seurcashondelivery/seurcashondelivery.php [hunk 16]
```diff
@@ -307,7 +306,36 @@ public function hookPaymentReturn($params)
...
+			
+			return (float)Tools::ps_round((float)(float)DB::getInstance()->getValue($sql), 2);
+		}
+		return 0;
+	}
+	public function getTotalPaid($order)
+	{
+		if ((int)$order->id >0)
+		{
+			$sql = 'SELECT total_paid FROM '._DB_PREFIX_.'seur_order WHERE id_order='.$order->id;
```
Comment (Quetzacoalt91) [inline install/seurcashondelivery/seurcashondelivery.php:324]: You cast the variable in your test, but not in the SQL request. Please add a `(int)` cast

File: install/seurcashondelivery/seurcashondelivery.php [hunk 17]
```diff
@@ -307,7 +306,36 @@ public function hookPaymentReturn($params)
...
+			
+			return (float)Tools::ps_round((float)(float)DB::getInstance()->getValue($sql), 2);
+		}
+		return 0;
+	}
+	public function updateTotalPaid($order,$total_paid)
+	{
+		if ((int)$order->id >0)
+		{
+			$sql = 'UPDATE '._DB_PREFIX_.'seur_order SET total_paid = '.$total_paid.' WHERE id_order='.$order->id;
```
Comment (Quetzacoalt91) [inline install/seurcashondelivery/seurcashondelivery.php:334]: And here again, please add a `(int)` cast.

File: controllers/admin/AdminSeur15Controller.php (hunk 14 above)
Comment (jcriadop) [inline controllers/admin/AdminSeur15Controller.php:41]: Line removed.

File: install/seurcashondelivery/seurcashondelivery.php (hunk 15 above)
Comment (jcriadop) [inline install/seurcashondelivery/seurcashondelivery.php:314]: Cast added.

File: install/seurcashondelivery/seurcashondelivery.php (hunk 16 above)
Comment (jcriadop) [inline install/seurcashondelivery/seurcashondelivery.php:324]: Cast added.

File: install/seurcashondelivery/seurcashondelivery.php (hunk 17 above)
Comment (jcriadop) [inline install/seurcashondelivery/seurcashondelivery.php:334]: Cast added.
